src/proofnetwork/xjdf/data/resource.cpp

```diff
@@ -0,0 +1,215 @@
+    }
+}
+
+void Resource::setAmountPool(const AmountPoolSP &arg)
+{
+    Q_D(Resource);
+    if (arg != d->amountPool) {
+        d->amountPool = arg;
+        emit amountPoolChanged(arg);
+    }
+}
+
+bool Resource::fillFromXJdf(QXmlStreamReader &)
+{
+    //NOTE: Nothing there for now, but it can be fill later
+    return false;
+}
+
+void Resource::readAttributesFromXJdf(QXmlStreamReader &reader)
+{
+    auto attributes = reader.attributes();
+    if (attributes.hasAttribute(QStringLiteral("ID"))) {
+        auto id = attributes.value(QStringLiteral("ID")).toString();
+        setId(id);
+    }
+    if (attributes.hasAttribute(QStringLiteral("Orientation"))) {
+        auto orientation = attributes.value(QStringLiteral("Orientation")).toString();
+        setOrientation(resourceOrientationFromString(orientation));
+    }
+}
+
+void Resource::toXJdf(QXmlStreamWriter &writer, bool writeEnd) const
+{
+    Q_D_CONST(Resource);
+    if (writeEnd) {
+        writer.writeEndElement();
+        return;
+    }
+
+    writer.writeStartElement(QStringLiteral("Resource"));
+    if (!d->id.isEmpty())
+        writer.writeAttribute(QStringLiteral("ID"), d->id);
+    if (d->orientation != ResourceOrientation::Rotate0Orientation)
+        writer.writeAttribute(QStringLiteral("Orientation"), resourceOrientationToString(d->orientation));
+    for (const auto &part : d->parts)
+        part->toXJdf(writer);
+    if (d->amountPool)
+        d->amountPool->toXJdf(writer);
+}
+
+ResourceSP Resource::fromXJdf(QXmlStreamReader &reader, const XJdfDocumentSP &document)
+{
+    ResourceSP resource;
+    QVector<PartSP> parts;
+    AmountPoolSP amountPool;
+    QString id;
+    ResourceOrientation orientation = ResourceOrientation::Rotate0Orientation;
+
+    if (reader.isStartElement() && reader.name() == QStringLiteral("Resource")) {
+        auto attributes = reader.attributes();
+        if (attributes.hasAttribute(QStringLiteral("ID")))
+            id = attributes.value(QStringLiteral("ID")).toString();
+        if (attributes.hasAttribute(QStringLiteral("Orientation")))
+            orientation = resourceOrientationFromString(attributes.value(QStringLiteral("Orientation")).toString());
+        reader.readNextStartElement();
+
+        while (!reader.atEnd() && !reader.hasError()) {
+            if (reader.isStartElement() && reader.name() == QStringLiteral("AmountPool")) {
+                amountPool = AmountPool::fromXJdf(reader, document);
+            } else if (reader.isStartElement() && reader.name() == QStringLiteral("Part")) {
+                auto part = Part::fromXJdf(reader, document);
+                if (part)
+                    parts << part;
+            } else if (reader.isStartElement()) {
+                auto creator = resourceCreator(reader.name().toString());
+                if (creator) {
+                    resource = creator(reader, document);
+                    reader.readNext();
+                    continue;
```

Comment:
shouldn't it be break? You will reset resource on each new start element otherwise and 164-165 lines don't do anything extra to just having creator called (because it will still do readNext() after ifs )